src/ripple/app/tx/impl/DepositPreauth.cpp

```diff
@@ -154,35 +156,50 @@ DepositPreauth::doApply ()
     }
     else
     {
-        // Verify that the Preauth entry they asked to remove is
-        // in the ledger.
         AccountID const unauth {ctx_.tx[sfUnauthorize]};
         uint256 const preauthIndex {getDepositPreauthIndex (account_, unauth)};
-        auto slePreauth = view().peek (keylet::depositPreauth (preauthIndex));
-
-        if (! slePreauth)
-        {
-            // Error should have been caught in preclaim.
-            JLOG(j_.warn()) << "Selected DepositPreauth does not exist.";
-            return tecNO_ENTRY;
-        }
 
-        auto viewJ = ctx_.app.journal ("View");
-        std::uint64_t const page {(*slePreauth)[sfOwnerNode]};
-        if (! view().dirRemove (
-            keylet::ownerDir (account_), page, preauthIndex, true))
-        {
-            JLOG(j_.warn()) << "Unable to delete DepositPreauth from owner.";
-            return tefBAD_LEDGER;
-        }
+        return removeDepositPreauthFromLedger (
+            ctx_.app, view(), preauthIndex, j_);
+    }
+    return tesSUCCESS;
+}
 
-        // If we succeeded, update the DepositPreauth owner's reserve.
-        auto const sleOwner = view().peek (keylet::account (account_));
-        adjustOwnerCount (view(), sleOwner, -1, viewJ);
+TER
+DepositPreauth::removeDepositPreauthFromLedger (Application& app,
+    ApplyView& view, uint256 const& preauthIndex, beast::Journal j)
+{
+    // Verify that the Preauth entry they asked to remove is
+    // in the ledger.
+    std::shared_ptr<SLE> const slePreauth {
+        view.peek (keylet::depositPreauth (preauthIndex))};
+    if (! slePreauth)
+    {
+        // Error should have been caught in preclaim.
```

Comment:
This is a static public function. I don't think `DepositPreauth::preclaim` is still a precondition to calling this. I'd probably just remove the comment